### PR TITLE
Add highlight-symbol

### PR DESCRIPTION
**Pull request by cg505**
This is a nice way to navigate through code that I use a lot, specifically `M-p` and `M-n`.

### CODE REVIEW

File: .emacs.d/lisp/code/code-hub.el [hunk 1]
```diff
@@ -3,6 +3,7 @@
 (kotct/hub "code"
            (magit-c
             indentation
+            code-navigation
```
Comment (rye) [inline .emacs.d/lisp/code/code-hub.el:6]: I think this file could best be renamed to just `navigation` (I will fix this shortly in a commit that I am about to push.)

File: .emacs.d/lisp/code/code-navigation.el [hunk 2]
```diff
@@ -0,0 +1,12 @@
+;;; M-n: go the next occurence of the symbol under point
+;;; M-p: go the previous occurence of the symbol under point
+;;;   note: these keybindings are enabled as a part of `highlight-symbol-nav-mode'
+;;; M-': replace all following occurences of the symbol under point
+
+;; enable highlight-symbol mode for all programming modes
+(add-hook 'prog-mode-hook (lambda ()
+                            (highlight-symbol-mode +1)
```
Comment (rye) [inline .emacs.d/lisp/code/code-navigation.el:8]: Is there any particular reason to use `+1` vs `t`?

File: .emacs.d/lisp/code/code-navigation.el [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+                            (highlight-symbol-nav-mode +1)
```
Comment (rye) [inline .emacs.d/lisp/code/code-navigation.el:9]: Is there any particular reason to use `+1` vs `t`?

**Review by rye — CHANGES_REQUESTED** [review]
Just a minor gripe with naming which I will take care of.

**Review by rye — APPROVED** [review]

File: .emacs.d/lisp/code/code-navigation.el (hunk 3 above)
Comment (cg505) [reply .emacs.d/lisp/code/code-navigation.el:9]: Yes. Modes enable on positive arguments and disable on non-positive arguments. To make this clear, it's a common practice to enable using +1 and disable using -1.

**Review by cg505 — COMMENTED** [review]
The reason I didn't do this was I wanted to leave open the feature name `navigation` in case we need it for a non-code version. That said, I'm find with just calling it `navigation` if you don't think it'll be a problem.

File: .emacs.d/lisp/code/code-navigation.el (hunk 3 above)
Comment (rye) [reply .emacs.d/lisp/code/code-navigation.el:9]: Ah, thanks for the clarification.

**Comment by rye** [comment]
Oh, I see your point.  I will revert that.  I think it's better to be clear.

**Comment by rye** [comment]
Looks good now.

**Comment by rye** [comment]
Merging.

**Pull request merged**